ntbs-service/Services/NotificationChangesService.cs

```diff
@@ -0,0 +1,283 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EFAuditer;
+using ntbs_service.DataAccess;
+using ntbs_service.Helpers;
+using ntbs_service.Models.Entities;
+using ntbs_service.Models.Entities.Alerts;
+using ntbs_service.Models.Enums;
+using ntbs_service.TagHelpers;
+using Serilog;
+
+namespace ntbs_service.Services
+{
+    public interface INotificationChangesService
+    {
+        Task<IEnumerable<NotificationHistoryListItemModel>> GetChangesList(int notificationId);
+    }
+
+    public class NotificationChangesService : INotificationChangesService
+    {
+        private readonly IAuditService _auditService;
+        private readonly IUserRepository _userRepository;
+        private Dictionary<string, string> UsernameDictionary { get; set; }
+
+        public NotificationChangesService(IAuditService auditService, IUserRepository userRepository)
+        {
+            _auditService = auditService;
+            _userRepository = userRepository;
+        }
+
+        public async Task<IEnumerable<NotificationHistoryListItemModel>> GetChangesList(int notificationId)
+        {
+            UsernameDictionary = await _userRepository.GetUsernameDictionary();
+
+            var auditLogs = (await _auditService.GetWriteAuditsForNotification(notificationId))
+                .OrderBy(log => log.AuditDateTime)
+                .GroupByConsecutive((prev, next) => (next.AuditDateTime - prev.AuditDateTime).TotalSeconds <= 2)
```

Comment:
seems like a magic number, why 2?

Reply:
Oh yeah, it is a very much number :S
It's a bit plucked out of the air tbh, something that large enoughensures that the audits created as part of a single audit action are grouped together, but small enough

Reply:
In fact, I might just cut it down to 1 second, feels a bit safer 🤔